File: surf-cloud-core/src/main/java/dev/slne/surf/cloud/core/netty/client/connection/TestInitlizier.java
package dev.slne.surf.cloud.core.netty.client.connection;

import dev.slne.surf.cloud.core.netty.protocol.channel.ChannelInitializerModifier;
import io.netty.channel.socket.SocketChannel;
import io.netty.handler.timeout.ReadTimeoutHandler;
import lombok.extern.flogger.Flogger;

@Flogger
//@Component
public class TestInitlizier implements ChannelInitializerModifier {

  @Override
  public void modify(SocketChannel ch) throws Exception {
    log.atInfo().log("Modifying channel");
    ch.pipeline().addLast(new ReadTimeoutHandler(5));
  }
}

File: surf-cloud-core/src/main/kotlin/dev/slne/surf/cloud/core/netty/client/connection/TestInitlizier.kt
package dev.slne.surf.cloud.core.netty.client.connection

import dev.slne.surf.cloud.api.util.logger
import dev.slne.surf.cloud.core.netty.protocol.channel.ChannelInitializerModifier
import io.netty.channel.Channel
import io.netty.handler.timeout.ReadTimeoutHandler

//@Component
class TestInitlizier : ChannelInitializerModifier {
    private val log = logger()

    override fun modify(ch: Channel) {
        log.atInfo().log("Modifying channel")
        ch.pipeline().addLast(ReadTimeoutHandler(5))
    }
}
